Util/main_script.py

```diff
@@ -0,0 +1,80 @@
+import scripts.dna_rna_tools as nk_tools
+import scripts.filter_fastq as fi_fa
+import os
+from typing import List, Union, Dict, Tuple
+
+
+def run_dna_rna_tools(*args: str) -> Union[str, List[str]]:
+    """
+    Executes the specified operation on the given sequences.
+    Args: Arbitrary number of strings with sequences,
+    the last argument should be the operation name
+    (transcribe, reverse, complement, reverse_complement, gc_cont, DNA_or_RNA).
+    Returns:
+    If one sequence is passed — str,
+    If multiple sequences are passed — list[str, str].
+    """
+    operation = args[- 1]
+    res = []
+    for i in range(len(args) - 1):
+        if operation == 'transcribe':
+            res.append(nk_tools.transcribe(args[i]))
+        elif operation == 'reverse':
+            res.append(nk_tools.reverse(args[i]))
+        elif operation == 'complement':
+            res.append(nk_tools.complement(args[i]))
+        elif operation == 'reverse_complement':
+            res.append(nk_tools.reverse_complement(args[i]))
+        elif operation == 'gc_cont':
+            res.append(nk_tools.gc_cont(args[i]))
+        elif operation == 'dna_or_rna':
+            res.append(nk_tools.dna_or_rna(args[i]))
```

Comment:
Вот тут можно также использовать словарь для сопоставления операций с функциями, чтобы избежать многочисленных if-elif и немного сократить код. Ну и можно еще добавить чтобы ошибка о неопознанной операции вылетала.  Хотя так код тоже весьма неплохо работает
```suggestion
   operations = {
    'transcribe': nk_tools.transcribe,
    'reverse': nk_tools.reverse,
    'complement': nk_tools.complement,
    'reverse_complement': nk_tools.reverse_complement,
    'gc_cont': nk_tools.gc_cont,
    'dna_or_rna': nk_tools.dna_or_rna,
}
if operation in operations:
    res = [operations[operation](seq) for seq in args[:-1]]
else:
    raise ValueError(f"Unsupported operation: {operation}")

```

Reply:
Не обязательно (Антону например больше нравится через услования) но, да, так можно. Как минимум полезно знать про разные варианты